core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -822,62 +822,23 @@ Cuda::size_type Cuda::device_arch() {
 void Cuda::impl_finalize() { Impl::CudaInternal::singleton().finalize(); }
 
 Cuda::Cuda()
-    : m_space_instance(&Impl::CudaInternal::singleton()), m_counter(nullptr) {
+    : m_space_instance(&Impl::CudaInternal::singleton(),
+                       [](Impl::CudaInternal *) {}) {
```

Comment:
Minor/non-blocking: since this is just instantiating a `std::function`, we could save on some lambda instantiations, etc., by initializing this with a `nullptr`. (Here and in `SYCL` and `HIP`, of course).